### PR TITLE
Rephrase paragraph on indicators vs indices and add links to their respective documentation

### LINKED ISSUE #1436: Move the discussion of indicator vs index to a more accessible location, and link to it on both pages for indicators and indices.
### Problem

While we've made great effort to clearly delineate the differences, it's still somewhat unclear to new users how these functions differ from each other. The notebooks and examples are good, but the information is not seen quickly enough when scanning the documentation.

### Potential Solution

The discussion of Indicator vs index should be more front and centre. This section needs to be pulled up so that users find it sooner. Links should be made between the sections via a `Notes` block before any indices/indicators are described.

On the README, we explain the most important factor being that `time`, but perhaps we can also explain here what the differences between them is and their intended audience (developers vs. end-users).

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1436 
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [ ] CHANGES.rst has been updated (with summary of main changes)
  - [ ] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Small documentation change related to indices vs indicators.

### Does this PR introduce a breaking change?
No

### Other information:

The original issue says the distinction between indicators and indices should be front and center, but it's already in the first paragraph of the Usage section, so not sure where it could go to get more visibility. Suggestions welcome.

### CODE REVIEW

File: docs/notebooks/usage.ipynb [hunk 1]
```diff
@@ -8,9 +8,7 @@
     "\n",
```
Comment (SarahG-579462) [inline docs/notebooks/usage.ipynb:8]: <p>Maybe make the sentence "Indicators should not be confused with <em>indices</em>, which define the algorithmic layer of each indicator." in bold? </p><p>Also, it might help the reader if this paragraph is split up into bullets/fragments. I know that for me, it's hard to parse what's the relevant information when it's so large.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Ouranosinc/xclim/pull/1472/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (SarahG-579462) [inline docs/notebooks/usage.ipynb:8]: <p>I think "Units Handling" is more appropriate than "Units handling", since "Units Handling" is a page with its own title, and also is a noun phrase, handling isn't just a normal participle</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Ouranosinc/xclim/pull/1472/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (huard) [reply docs/notebooks/usage.ipynb:8]: Done

**Review by SarahG-579462 — APPROVED** [review]
Looks good!
